src/simulator.zig

```diff
@@ -295,6 +295,8 @@ pub fn main() !void {
             stdx.unimplemented("repair requires reachable primary");
         } else if (simulator.core_missing_quorum()) {
             output.warn("no liveness, core replicas cannot view-change", .{});
+        } else if (simulator.core_missing_checkpoint()) {
+            output.warn("no liveness, core can't find canonical checkpoint", .{});
```

Comment:
Nit, but should this come after we check `core_missing_prepare` and `core_missing_blocks`? Since if the cluster is missing a prepare/block, that could cause us to be unable to replay a particular checkpoint.

Reply:
Ah, that's actually a good point. This is intentionally before we check the blocks, because, at the commit where we observed the failure originally, `core_missing_blocks` checks for a stronger property, that all core replicas are missing the same _set_ of blocks (which is not true if they are on different checkpoints).

But we've since switched to a weaker property, where we just check that a block isn't present in the core.

Reply:
moved it to be the last check, as we _do_ want to check that all blocks are repaired even if there isn't a canonical checkpoint